### PR TITLE
Commit vendor to the repo

### PR DESCRIPTION
**Pull request by randomvariable**
**What this PR does / why we need it**:
Commits the vendor directory as per https://github.com/kubernetes-sigs/cluster-api-provider-aws/pull/283#discussion_r226321180

Development instructions are updated to explicitly call out running `dep ensure` and `gazelle`.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
In support of #13 

**Special notes for your reviewer**:

_Please confirm that if this PR changes any image versions, then that's the sole change this PR makes._

This  PR is guaranteed to break your browser

**Release note**:

```release-note

```

### CODE REVIEW

**Comment by detiber** [comment]
@liztio this overlaps with your bazel PR, could you help reconcile any diff?

File: Makefile [hunk 1]
```diff
@@ -21,14 +21,11 @@ GOPATH := $(shell go env GOPATH)
...
 
-ifndef FASTBUILD
-.PHONY: vendor
-endif
-
 # Dependency managemnt.
 vendor:
 	dep version || go get -u github.com/golang/dep/cmd/dep
 	dep ensure
+	bazel run gazelle
```
Comment (liztio) [inline Makefile]: it's `//:gazelle` I think

**Comment by randomvariable** [comment]
Rebased with #279

File: docs/development.md [hunk 2]
```diff
@@ -23,6 +23,14 @@ When you update the mocks, please make sure the imports are not from the depende
 
 For example, if you see `types "sigs.k8s.io/cluster-api/vendor/k8s.io/apimachinery/pkg/types"` in the import path, replace it with `types "k8s.io/apimachinery/pkg/types"`.
 
+## Committing code
+Ensure you have updated the vendor directory with:
+
+```
+make vendor REVENDOR=y
```
[the remark this answers is not in the document]
Comment (randomvariable) [reply docs/development.md]: There's the bazel step as well at the moment, so it'd be `dep ensure && bazel run ://gazelle`

File: docs/development.md [hunk 3]
```diff
@@ -36,8 +44,9 @@ storage.
    - `brew install gettext && brew link --force gettext` on MacOS.
 1. Install [minikube][minikube]
    - `brew install minikube` on MacOS.
+1. Install [bazel][bazel]
 1. Configure `minikube`:
-    1. Use Kubernetes v1.9.4 `minikube config set kubernetes-version v1.9.4`.
+    1. Use Kubernetes v1.9.4 `minikube config set kubernetes-version v1.12.1`.
```
[the remark this answers is not in the document]
Comment (randomvariable) [reply docs/development.md]: Fixed, thanks.

**Pull request merged**